### DISCUSSION TITLE
Integrate Clay model weights with SAM

### DISCUSSION THREAD

**Discussion by CryptoRhinoGH** [Clay-foundation/model#329] [Q&A]
Hey guys! Working with Segment anything model but trying to use sentinel 2 images. I'm trying to instead use weights for Clay model to generate the embeddings to be used with SAM. Can anyone help me understand if this is possible, and if so where are the weights located which I can import into SAM?

For example, the following ipynb uses the sam_vit_h_4b8939.pth file as the checkpoint. I'm wondering if its possible to use the Clay Model weights in such a scenario. My goal is to use the superior power of Clay model's weights and its training on sentinel 2 L2A data.
https://github.com/facebookresearch/segment-anything/blob/main/notebooks/predictor_example.ipynb

**Comment by rbavery** [comment]
This is not possible without jointly training the SAM decoder and Clay Encoder
